gdrive/analytics_api.py

```diff
@@ -0,0 +1,166 @@
+"""
+Google Analytics Rest API
+"""
+
+from datetime import datetime, timedelta
+import logging
+
+import fastapi
+from pydantic import BaseModel
+from fastapi import BackgroundTasks, responses
+import pandas as pd
+
+from gdrive import error, settings, analytics_client, sheets_client, drive_client
+
+log = logging.getLogger(__name__)
+router = fastapi.APIRouter()
+
+
+class AnalyticsRequest(BaseModel):
+    startDate: str = None
+    endDate: str = None
+
+
+@router.post("/analytics")
+async def run_analytics_default():
+    # Default behaviour for the system is run a report from previous day
+    target_date = datetime.today() - timedelta(days=1)
+    run_analytics(target_date, None)
+    return responses.JSONResponse(
+        status_code=202,
+        content="Analytics report for %s complete." % (datetime.date(datetime.today())),
+    )
+
+
+@router.post("/analytics/daterange")
```

Comment:
does this need to be a serperate endpoint? can it just default to 1 day if no date sent?